### ISSUE TITLE
Image published to Harbor registry doesn't show up when listing images

### ISSUE THREAD

**Issue by trisberg** [buildpack/pack#100]
I have a Harbor registry running and I can build and publish images with pack but the registry listing of images do not include the new image. I can pull it and push it with docker and then I can see the image in the listing.

I have a simple node app that I build to `harbor.springdeveloper.com/test/node-app`

```
pack build --path . --publish harbor.springdeveloper.com/test/node-app
Using default builder image packs/samples:v3alpha2
Pulling builder image packs/samples:v3alpha2 (use --no-pull flag to skip this step)
v3alpha2: Pulling from packs/samples
Digest: sha256:9070b18718776e5a45f5cb91376e9fad3f4f82815d82dc7895a164b9fb6df180
Status: Image is up to date for packs/samples:v3alpha2
Selected run image packs/run:v3alpha2 from builder packs/samples:v3alpha2
Using cache volume pack-cache-b90667b76738b2549a46e8389ae2f595
===> DETECTING
[detector] 2019/02/20 16:10:17 Trying group of 1...
[detector] 2019/02/20 16:10:17 ======== Results ========
[detector] 2019/02/20 16:10:17 Sample Node.js Buildpack: pass
===> ANALYZING

...

[exporter] *** Image: harbor.springdeveloper.com/test/node-app@sha256:d7d91353f0bf42279f8f131923683efbe7a5b74ebd0d2e9dbdfde16a4c9821ef
Successfully built image harbor.springdeveloper.com/test/node-app
```

Once it completes I list the images in the `test` project (project_id is 5)

```
curl -X GET -L -w '\n' -H 'Accept: application/json' 'http://harbor.springdeveloper.com/api/repositories?project_id=5'
[]
```

I can pull and then push the image:
```
$ docker pull harbor.springdeveloper.com/test/node-app
Using default tag: latest
latest: Pulling from test/node-app
38e2e6cd5626: Already exists 
705054bc3f5b: Already exists 
c7051e069564: Already exists 
7308e914506c: Already exists 
2da8bfb3b5b7: Already exists 
4c51b7d9e9ea: Already exists 
80e339c0631f: Already exists 
01e734c85049: Already exists 
09b3ec4f3bed: Already exists 
2501aaff82c2: Already exists 
a41c740297ab: Pull complete 
f60b027f8900: Pull complete 
b76f64a78498: Pull complete 
ae01127db166: Pull complete 
47f92320f424: Pull complete 
Digest: sha256:d7d91353f0bf42279f8f131923683efbe7a5b74ebd0d2e9dbdfde16a4c9821ef
Status: Downloaded newer image for harbor.springdeveloper.com/test/node-app:latest

$ docker push harbor.springdeveloper.com/test/node-app
The push refers to repository [harbor.springdeveloper.com/test/node-app]
dc0edb505e31: Layer already exists 
302618f06239: Layer already exists 
d77dc7ed6207: Layer already exists 
f681c048b25a: Layer already exists 
30e0fe77ff05: Layer already exists 
fa4d876615d8: Layer already exists 
b68f13611ea3: Layer already exists 
72a6570c4422: Layer already exists 
3210aa063b89: Layer already exists 
b2fedec7b904: Layer already exists 
0e84b32bb1dd: Layer already exists 
27a216ffe825: Layer already exists 
9e9d3c3a7458: Layer already exists 
7604c8714555: Layer already exists 
adcb570ae9ac: Layer already exists 
latest: digest: sha256:389e587e6031e24948af1a674c32a257b979edab1f0ebc68d7d43b9e8d10bfac size: 3454
```
And now the image does show up:
```
$ curl -X GET -L -w '\n' -H 'Accept: application/json' 'http://harbor.springdeveloper.com/api/repositories?project_id=5'
[
  {
    "id": 54,
    "name": "test/node-app",
    "project_id": 5,
    "description": "",
    "pull_count": 0,
    "star_count": 0,
    "tags_count": 2,
    "labels": [],
    "creation_time": "2019-02-20T16:13:14.539166Z",
    "update_time": "2019-02-20T16:13:14.539166Z"
  }
]
```

Complete build log:
```
$ pack build --path . --publish harbor.springdeveloper.com/test/node-app
Using default builder image packs/samples:v3alpha2
Pulling builder image packs/samples:v3alpha2 (use --no-pull flag to skip this step)
v3alpha2: Pulling from packs/samples
Digest: sha256:9070b18718776e5a45f5cb91376e9fad3f4f82815d82dc7895a164b9fb6df180
Status: Image is up to date for packs/samples:v3alpha2
Selected run image packs/run:v3alpha2 from builder packs/samples:v3alpha2
Using cache volume pack-cache-b90667b76738b2549a46e8389ae2f595
===> DETECTING
[detector] 2019/02/20 16:10:17 Trying group of 1...
[detector] 2019/02/20 16:10:17 ======== Results ========
[detector] 2019/02/20 16:10:17 Sample Node.js Buildpack: pass
===> ANALYZING
Reading information from previous image for possible re-use
[analyzer] 2019/02/20 16:10:18 WARNING: image 'harbor.springdeveloper.com/test/node-app' not found or requires authentication to access
[analyzer] 2019/02/20 16:10:18 using cached layer 'io.buildpacks.samples.nodejs/{{/workspace/io.buildpacks.samples.nodejs/npm-cache io.buildpacks.samples.nodejs:npm-cache}}'
[analyzer] 2019/02/20 16:10:18 removing stale cached launch layer 'io.buildpacks.samples.nodejs:{{/workspace/io.buildpacks.samples.nodejs/node_modules io.buildpacks.samples.nodejs:node_modules}}', not in metadata 
[analyzer] 2019/02/20 16:10:18 removing stale cached launch layer 'io.buildpacks.samples.nodejs:{{/workspace/io.buildpacks.samples.nodejs/nodejs io.buildpacks.samples.nodejs:nodejs}}', not in metadata 
===> BUILDING
[builder] npm notice created a lockfile as package-lock.json. You should commit this file.
[builder] npm WARN app@1.0.0 No description
[builder] npm WARN app@1.0.0 No repository field.
[builder] 
[builder] added 48 packages from 36 contributors and audited 121 packages in 1.971s
[builder] found 0 vulnerabilities
[builder] 
===> EXPORTING
[exporter] 2019/02/20 16:10:29 adding layer 'app' with diffID 'sha256:30e0fe77ff0566d190cfa24c20e3d39c96a3dcb54f9d29b7998df8a7e72e112e'
[exporter] 2019/02/20 16:10:29 adding layer 'config' with diffID 'sha256:f681c048b25ac9b2e563323c76dd01cbee59584790281b88adfe886c283641a2'
[exporter] 2019/02/20 16:10:29 adding layer 'launcher' with diffID 'sha256:d77dc7ed6207d6bb9c389aa5f087ea7fffea9238e2de84b03f8b3c1152e1e58f'
[exporter] 2019/02/20 16:10:29 adding layer 'io.buildpacks.samples.nodejs:node_modules' with diffID 'sha256:302618f06239200810a871ffc701a367cf9ce718838c021cd5c7e11d881f29b8'
[exporter] 2019/02/20 16:10:30 adding layer 'io.buildpacks.samples.nodejs:nodejs' with diffID 'sha256:dc0edb505e31c571e8a5615be06628ce497544688a2c8a471f4f56aa0b5445c5'
[exporter] 2019/02/20 16:10:31 setting metadata label 'io.buildpacks.lifecycle.metadata'
[exporter] 2019/02/20 16:10:31 setting env var 'PACK_LAYERS_DIR=/workspace'
[exporter] 2019/02/20 16:10:31 setting env var 'PACK_APP_DIR=/workspace/app'
[exporter] 2019/02/20 16:10:31 setting entrypoint '/lifecycle/launcher'
[exporter] 2019/02/20 16:10:31 setting empty cmd
[exporter] 2019/02/20 16:10:31 writing image
[exporter] 2019/02/20 16:10:31 existing blob: sha256:80e339c0631fb56471f967a9838454f62212fcf097b3424152b3a2cd266f52aa
[exporter] 2019/02/20 16:10:31 existing blob: sha256:38e2e6cd5626f31cea1a0a5751a9a7e6564c589a3388dcf84f00d4bb98146844
[exporter] 2019/02/20 16:10:31 existing blob: sha256:2da8bfb3b5b741820e6cb29a71f66e8d944ddad2d00c6dc1e31461120ed39e46
[exporter] 2019/02/20 16:10:31 existing blob: sha256:7308e914506c09fa6d2242368545f55462e024f785e21b21b3e90403081a9336
[exporter] 2019/02/20 16:10:31 existing blob: sha256:47f92320f424b57b3beea8a0bcca978cfb03ed1cd5426348ce697d79901a0f64
[exporter] 2019/02/20 16:10:31 existing blob: sha256:4c51b7d9e9ea44f5e1a432b04fd1d3a2accbcc8e8c2ba7455155a6661c868fb8
[exporter] 2019/02/20 16:10:31 existing blob: sha256:b76f64a78498234c109bc1addf29af70508b0f478ed545f7178410b37ef8a0bb
[exporter] 2019/02/20 16:10:31 existing blob: sha256:c7051e0695642c5bf01467a4a64106cabfa20e57fee91b31ee73f7308bccf5bd
[exporter] 2019/02/20 16:10:31 existing blob: sha256:c48fdf54b9802d5a7466d40a4b65e34914eaabe9b77002f192cf60fe6c4b0372
[exporter] 2019/02/20 16:10:31 existing blob: sha256:705054bc3f5bd722eb8f026532447fac897c521d8906eb36e79a60d0fb0606fa
[exporter] 2019/02/20 16:10:31 existing blob: sha256:2501aaff82c2d492dc3f4fd4f526fd3c2e7fd090297559566733f170d92d692f
[exporter] 2019/02/20 16:10:31 existing blob: sha256:ae01127db166cbaa431449803efe95613b9eec68e133d0cb804c4a8f56f3f0d3
[exporter] 2019/02/20 16:10:31 existing blob: sha256:01e734c850498f929b8fb427f845545d3ebf3165afdb77c85a9f7dcbda1adc60
[exporter] 2019/02/20 16:10:31 existing blob: sha256:09b3ec4f3bede08b94d415ed2318e8766fb189bf31e69c9ed5d5db5055982f07
[exporter] 2019/02/20 16:10:31 existing blob: sha256:f60b027f89003a22a71284556221c201670b60545027a09228f02339626c4fad
[exporter] 2019/02/20 16:10:31 existing blob: sha256:a41c740297abac35b27b2e4a67c3923af755abbb918015ddbf7e202437f8297c
[exporter] 2019/02/20 16:10:31 harbor.springdeveloper.com/test/node-app:latest: digest: sha256:d7d91353f0bf42279f8f131923683efbe7a5b74ebd0d2e9dbdfde16a4c9821ef size: 2696
[exporter] 2019/02/20 16:10:31 
[exporter] *** Image: harbor.springdeveloper.com/test/node-app@sha256:d7d91353f0bf42279f8f131923683efbe7a5b74ebd0d2e9dbdfde16a4c9821ef
Successfully built image harbor.springdeveloper.com/test/node-app
```

**Comment by ekcasey** [comment]
@trisberg It looks like it is an issue with Harbor itself. Harbor has a set of whitelisted user-agents and it will not record a "push" event in the db unless it is coming from the docker client. See this jib issue for context https://github.com/goharbor/harbor/issues/5729

It appears to be fixed in master of Harbor (https://github.com/goharbor/harbor/pull/6689), but the release notes don't indicate that any of the newest releases have the fix yet.

(@sclevine I think you will find this discussion interesting)

**Comment by trisberg** [comment]
Thanks for tracking this down. Looks like the fix will be in next 1.8 release in a few months and not ported to any 1.7.x releases (https://github.com/goharbor/harbor/issues/7056#issuecomment-468890510).

**Comment by trisberg** [comment]
New plan, looks like this will be fixed in the next 1.7.5 release (https://github.com/goharbor/harbor/issues/7056#issuecomment-478881025)
